SignalUtilitiesKit/Database/Migrations/BlockingManagerRemovalMigration.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2022 Rangeproof Pty Ltd. All rights reserved.
+
+import Foundation
+
+@objc(SNBlockingManagerRemovalMigration)
+public class BlockingManagerRemovalMigration: OWSDatabaseMigration {
+    @objc
+    class func migrationId() -> String {
+        return "003"    // FIXME: MigrationId needs to be updated based on other PRs
+    }
+
+    override public func runUp(completion: @escaping OWSDatabaseMigrationCompletion) {
+        self.doMigrationAsync(completion: completion)
+    }
+
+    private func doMigrationAsync(completion: @escaping OWSDatabaseMigrationCompletion) {
+        // These are the legacy keys that were used to persist the "block list" state
+        let kOWSBlockingManager_BlockListCollection: String = "kOWSBlockingManager_BlockedPhoneNumbersCollection"
+        let kOWSBlockingManager_BlockedPhoneNumbersKey: String = "kOWSBlockingManager_BlockedPhoneNumbersKey"
+        
+        let dbConnection: YapDatabaseConnection = primaryStorage.newDatabaseConnection()
+        
+        let blockedSessionIds: Set<String> = Set(dbConnection.object(
+            forKey: kOWSBlockingManager_BlockedPhoneNumbersKey,
+            inCollection: kOWSBlockingManager_BlockListCollection
+        ) as? [String] ?? [])
+
+        Storage.write(
+            with: { transaction in
+                Storage.shared.getAllContacts(with: transaction)
+                    .filter { contact -> Bool in blockedSessionIds.contains(contact.sessionID) }
+                    .forEach { contact in
+                        contact.isBlocked = true
+                        Storage.shared.setContact(contact, using: transaction)
+                    }
+                
+                // Now that the values have been migrated we can clear out the old collection
+                transaction.removeAllObjects(inCollection: kOWSBlockingManager_BlockListCollection)
+                
+                self.save(with: transaction) // Intentionally capture self
+            },
+            completion: {
+                // FIXME: Need to force a config sync after this migration
```

Comment:
@RyanRory Would be good to get your thoughts on this - We should force a config message after this migration is run but this file is in `SignalUtilitiesKit` so we don't have access to do so.

What do you think about forcing a config message whenever any migration has actually been run? (other option I can think of is to try and pass a flag indicating the migration needs a config sync back through the various completion blocks 😕)

Reply:
Ended up passing the flag through the migration system, also shifted the force configuration logic to a static MessageSender function so it doesn't need to be called from the main thread or via the `AppDelegate` anymore